app/controllers/orders_controller.rb

```diff
@@ -2,17 +2,14 @@ class OrdersController < ApplicationController
   def new
     @order = Order.new
     cart = Cart.find_by(session_key: session.id.to_s)
-    @order_products = cart.cart_products
-    @total_amount = cart.total_amount
+    @order.create_from_cart(cart)
```

Comment:
It would be better to have `create_from_cart` on the class side. In this case it could be refactored to:

``` ruby
cart = Cart.find_by(session_key: session.id.to_s)
@order = Order.create_from_cart(cart)
```